### PR TITLE
Add TranslatableType

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Extracted some changes from https://github.com/PrestaShop/PrestaShop/pull/10408
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | In forms "Shop parameters > Product Settings", "Orders > Invoices" and "Orders > Delivery Slips" translatable inputs were updated but they should keep working as they were before. So the QA test is to check the behavior is unchanged although the code has been updated.

This is how it looks when used in form:

translatable_type

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10704)

### CODE REVIEW

**Comment by sarjon** [comment]
@matks can you review?

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
@sarjon 👍 

That was a great candidate for PR splitting 😉 

I'm sorry you need to rebase in order to make Travis build pass as Travis build have been fixed by https://github.com/PrestaShop/PrestaShop/pull/10716

**Comment by PierreRambaud** [comment]
No need to rebase :) We need to merge 1.7.5.x into develop first =)

**Comment by sarjon** [comment]
> No need to rebase :)

already did. :D

**Comment by matks** [comment]
@PierreRambaud yes indeed :)
sorry for the mistake @sarjon we only needed to relaunch travis build

**Review by jolelievre — CHANGES_REQUESTED** [review]
I don't understand the purpose of this new type, what is the difference with TranslateType or TranslateTextType?
is the purpose to have a generic TranslatableType in which you can inject any other FormType?

**Comment by sarjon** [comment]
> I don't understand the purpose of this new type

@jolelievre as far as i know `TranslateType` is used in Product page and changing it could break it. Another issue is that `TranslateType` uses tabs instead of dropdowns to select language, for example: https://prnt.sc/kzfbbi

regarding `TranslateTextType` it should be depracated in favor of this new type as you can do the same (and even more) with it.

**Comment by jolelievre** [comment]
Ok, I didn't know `TranslateType` was displayed with tabs, I figured it was like `TranslateTextType`
The you're right we should deprecate the former type, which means adding a throw a deprecation notice and add the appropriate annotation
And of course replace it everywhere it's been used so far

**Review by jolelievre — APPROVED** [review]

**Review by jolelievre — DISMISSED** [review]
We should deprecate the form `TranslateTextType` by:
- adding a @deprecated annotation on it
- add this code in the constructor 
```php
@trigger_error('The '.__NAMESPACE__.'\TranslateTextType class is deprecated since version 1.7.6 and will be removed in 1.8. Use the '.__NAMESPACE__.'\TranslatableType class instead.', E_USER_DEPRECATED);
```
- switch the two types everywhere

**Comment by sarjon** [comment]
@jolelievre done.

File: src/PrestaShopBundle/Form/Admin/Type/TranslateTextType.php [hunk 1]
```diff
@@ -35,9 +35,23 @@
 
 /**
  * Class is responsible for creating translatable text inputs.
+ *
+ * @deprecated The since version 1.7.6 and will be removed in 1.8. Use the TranslatableType instead.
```
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Type/TranslateTextType.php]: @sarjon just a little typo problem here, the rest is perfect Good job!

Comment (sarjon) [reply src/PrestaShopBundle/Form/Admin/Type/TranslateTextType.php]: :man_facepalming: thanks, done.

**Comment by sarjon** [comment]
@jolelievre @matks is it possible to merge it today? ^^

**Comment by matks** [comment]
@sarjon I am surprised you were able to replace the usage of `TranslateType` so easily 😮 

Are you confident ? It looks "too nice" to be true ^^ ?

**Comment by sarjon** [comment]
@matks yup, i tried to make it as simple as possible, but it would be nice to have QA validate it. :)

**Review by matks — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
nice job @sarjon

**Comment by Quetzacoalt91** [comment]
Thank you @sarjon

**Pull request merged**